papermill_origami/noteable_dagstermill/assets.py

```diff
@@ -0,0 +1,288 @@
+
+        with tempfile.TemporaryDirectory() as output_notebook_dir:
+            with safe_tempfile_path() as output_log_path:
+
+                prefix = str(uuid.uuid4())
+                parameterized_notebook_path = os.path.join(
+                    output_notebook_dir, f"{prefix}-inter.ipynb"
+                )
+
+                executed_notebook_path = os.path.join(output_notebook_dir, f"{prefix}-out.ipynb")
+
+                # Scaffold the registration here
+                nb = load_notebook_node(notebook_path)
+                compute_descriptor = "op"
+
+                base_parameters = get_papermill_parameters(
+                    step_execution_context,
+                    inputs,
+                    output_log_path,
+                    compute_descriptor,
+                )
+                job_definition_id = step_execution_context.job_name
+                job_instance_id = step_execution_context.run_id
+                file_id = removeprefix(notebook_path, "noteable://")
+                noteable_parameters = {
+                    'job_definition_id': job_definition_id,
+                    'job_instance_id': job_instance_id,
+                    'file_id': file_id,
+                }
+                context_args = base_parameters["__dm_context"]
+                pipeline_context_args = dict(
+                    executable_dict=base_parameters["__dm_executable_dict"],
+                    pipeline_run_dict=base_parameters["__dm_pipeline_run_dict"],
+                    solid_handle_kwargs=base_parameters["__dm_solid_handle_kwargs"],
+                    instance_ref_dict=base_parameters["__dm_instance_ref_dict"],
+                    step_key=base_parameters["__dm_step_key"],
+                    **context_args,
+                )
+                # This is needed in order to set the pipeline context on the `_MANAGER_FOR_NOTEBOOK_INSTANCE`,
+                # which is then used by _load_input_parameter to fetch input values.
+                _reconstitute_pipeline_context(**pipeline_context_args)
+                load_input_template = "cloudpickle.loads(b64decode({serialized_val}))"
+                input_params_list = [
+                    PythonTranslator().codify(noteable_parameters, "Noteable provided parameters"),
+                    "\n# Dagster provided parameter inputs\n",
+                    "\n".join(
+                        [
+                            f"{input_name} = {load_input_template.format(serialized_val=b64encode(pickle.dumps(_load_input_parameter(input_name))))}"  # noqa: E501
+                            for input_name in base_parameters["__dm_input_names"]
+                        ]
+                    ),
+                ]
+                input_parameters = "".join(input_params_list)
+
+                template = f"""# Injected parameters
+import cloudpickle
+from base64 import b64decode
+{input_parameters}
+"""
```

Comment:
@jamiedemaria I must have missed this earlier but, was there a reason why we did not include the pipeline `context` construction code here?